apps/src/templates/teacherDashboard/teacherSectionsRedux.js

```diff
@@ -60,17 +76,68 @@ export default function teacherSections(state=initialState, action) {
   }
 
   if (action.type === SET_SECTIONS) {
+    const assignmentList = assignments(state);
+    const sections = action.sections.map(section =>
+      sectionFromServerSection(section, assignmentList, state.studioUrl));
     return {
       ...state,
-      sectionIds: action.sections.map(section => section.id),
-      sections: _.keyBy(action.sections, 'id')
+      sectionIds: sections.map(section => section.id),
+      sections: _.keyBy(sections, 'id')
+    };
+  }
+
+  // TODO: write tests for this
+  if (action.type === UPDATE_SECTION) {
+    const assignmentList = assignments(state);
+    const section = sectionFromServerSection(action.serverSection,
+      assignmentList, state.studioUrl);
+
+    return {
+      ...state,
+      sections: {
+        ...state.sections,
+        [action.sectionId]: {
+          ...state.sections[action.sectionId],
+          ...section
+        }
+      }
     };
   }
 
   return state;
 }
 
-// Selectors
+// Helpers and Selectors
+
+/**
+ * Maps from the data we get back from the server for a section, to the format
+ * we want to have in our store.
+ */
+// TODO(bjvanminnen): write some tests for this
+const sectionFromServerSection = (serverSection, assignmentList, studioUrl) => {
+  const courseId = serverSection.course_id || null;
+  const scriptId = serverSection.script ? serverSection.script.id : null;
+
+  const assignmentIndex = getAssignmentIndex(assignmentList)(courseId, scriptId);
+  const assignment = assignmentList[assignmentIndex];
+
+  return {
+    id: serverSection.id,
+    name: serverSection.name,
+    loginType: serverSection.login_type,
+    grade: serverSection.grade,
+    stageExtras: serverSection.stage_extras,
+    pairingAllowed: serverSection.pairing_allowed,
+    numStudents: serverSection.students.length,
+    code: serverSection.code,
+    courseId: serverSection.course_id,
+    scriptId: scriptId,
+    // TODO(bjvanminnen): should maybe be getting these fields as selectors instead of
+    // living in state
+    assignmentName: assignment ? assignment.name : '',
+    assignmentPath: assignment ? (studioUrl + getPath(assignment)) : ''
```

Comment:
I don't think these should live in our store. Instead we should have something like `getAssignmentName(state, sectionId)` and the same for path.